staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -786,29 +792,45 @@ type WatchListResult struct {
 	// the end of the stream.
 	initialEventsEndBookmarkRV string
 
-	// gv represents the API version
-	// it is used to construct the final list response
-	// normally this information is filled by the server
-	gv schema.GroupVersion
+	// negotiatedObjectDecoder knows how to decode
+	// the initialEventsListBlueprint
+	negotiatedObjectDecoder runtime.Decoder
+
+	// base64EncodedInitialEventsListBlueprint contains an empty,
+	// versioned list encoded in the requested format
+	// (e.g., protobuf, JSON, CBOR) and stored as a base64-encoded string
+	base64EncodedInitialEventsListBlueprint string
 }
 
 func (r WatchListResult) Into(obj runtime.Object) error {
```

Comment:
For my senerio, I'm trying to impliment Table decoding and I need to use `WatchList` to replace regular `Get`. But there is a promblem here, the regular `Get` would be able to return an `runtime.Object` which follows the result from `decoder`, but currently the type of argument pass to `Into` have to static.

https://github.com/kubernetes/kubernetes/blob/fb33a50816b03e7b484a5452dea96cb601630868/staging/src/k8s.io/client-go/rest/request.go#L1561-L1586

for example, if `k8s.io/cli0runtime/pkg/resource.UnstructuredPlusDefaultContentConfig` is configured as the decoder, `Get` may return both `unstructured.Unstructured` or `unstructured.UnstructuredList`, which depends on the whether has an Items field in response.

I created an e2e to describe this better:
1. decoded and return an `Unstructured`: https://github.com/kubernetes/kubernetes/pull/127927/commits/fb33a50816b03e7b484a5452dea96cb601630868#diff-2a4a79d31f8a4ac5523a2237bc54298fa8278a87383bff23457ae5db7e874173R268
2. decoded and return an `UnstructuredList`: https://github.com/kubernetes/kubernetes/pull/127927/commits/fb33a50816b03e7b484a5452dea96cb601630868#diff-2a4a79d31f8a4ac5523a2237bc54298fa8278a87383bff23457ae5db7e874173R306

maybe we need two methods here, just like `runtime.Decode` and `runtime.DecodeInto` does. The previous one returns a new object, and the later one guarantee given object is fulfilled.

see my poc here: https://github.com/kubernetes/kubernetes/pull/127927/commits/fb33a50816b03e7b484a5452dea96cb601630868#diff-e6b16db60c379a7e5fdb8f46054d9a87ea511dba3f94d261b5f61a506fcb5be2R815